app/templates/admin/services/upsert.html

```diff
@@ -0,0 +1,160 @@
+        <div class="mb-3">
+          <label class="form-label">Name</label>
+          <input id="name" type="text" class="form-control" name="name" value="{{ service["name"] if service else '' }}">
+        </div>
+
+        <div class="mb-3">
+          <label class="form-label">Description</label>
+          <textarea id="description" name="description" class="form-control" rows="3">{{ service["description"] if service else '' }}</textarea>
+        </div>
+
+        <div class="mb-3">
+          <label class="form-label">Images</label>
+          <input name="images" multiple="multiple" class="form-control" type="file" id="images">
+        </div>
+
+        <div class="mb-3" id="priceRows">
+          <label class="form-label">
+            Price
+          </label>
+          <button id="addPriceRow" type="button" class="btn btn-sm btn-link float-end">
+            Add Price
+          </button>
+          {% if service  %}
+            {% for price in service["price"] %}
+              <div class="input-group mb-1">
+                <input value="{{ price[0] }}" class="price-row form-control" type="text" placeholder="Name..."/>
+                <span class="input-group-text" id="addon-wrapping"><i class="fas fa-dollar-sign"></i></span>
+                <input value="{{ price[1] }}" class="price-row form-control" type="text" placeholder="Price..."/>
+                <button type="button" class="price-del-row btn btn-danger"><i class="fas fa-trash"></i></button>
+              </div>
+            {% endfor %}
+          {% endif %}
+        </div>
+        <div class="d-grid gap-2">
+          <button class="btn btn-primary full" type="submit">Save</button>
+        </div>
+      </form>
+    </div>
+    <div class="col">
+      {% if service %}
+        <label class="form-label">
+          Uploaded Images
+        </label>
+        {% for image in service.images %}
+          <div class="mb-3">
+            <img src="/static/images/{{ image }}" class="img-thumbnail" title="{{ image }}"/>
+            <div class="d-grid">
+              <button data-img="{{ image }}" class="price-rm-img btn btn-sm btn-secondary full" type="submit">Remove</button>
+            </div>
+          </div>
+        {% endfor %}
+      {% endif %}
+    </div>
+  </div>
+  <br><br>
+{% endblock %}
+
+{% block js %}
+  <script>
+    const addService = (payload) => {
+      $.ajax({
+        url: `/admin/services/adding`,
+        type: "POST",
+        data: payload,
+        contentType: false,
+        processData: false
+      }).done((resp) => {
+        location.reload();
+      })
+    };
+
+    const editService = (serviceId, payload) => {
+      $.ajax({
+        url: `/admin/services/${serviceId}/editing`,
+        type: "POST",
+        data: payload,
+        contentType: false,
+        processData: false
+      }).done((resp) => {
+        location.reload();
+      })
+    };
+
+    $(document).ready(() => {
```

Comment:
can we take this section (javascript section) into a separate file? and just load it here in the HTML page

Reply:
yes sure we can do it